src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java

```diff
@@ -105,6 +118,14 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
             IPythonUserConfig jobUserConfig = new IPythonUserConfig(kernel, launchTimeout, maxResults);
             // Get the right channel to execute the code
             run.setResult(launcher.getChannel().call(new ExecutorImpl(ws, listener, jobUserConfig)));
+            ResultAction a = run.getAction(ResultAction.class);
+            // search and update for action after the build
+            if (a != null) {
+                run.replaceAction(a.updateSummary());
+            } else {
+                run.addAction(new ResultAction(run, ws));
+            }
```

Comment:
@Loghijiaha , out of curiosity, could we just do something like `run.addOrReplaceAction(new ResultAction(run, ws));`?

Reply:
Yes that works @kinow, Thanks for teh catch.